server/channels/api4/status.go

```diff
@@ -106,7 +108,11 @@ func updateUserStatus(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	currentStatus, err := c.App.GetStatus(c.Params.UserId)
 	if err == nil && currentStatus.Status == model.StatusOutOfOffice && status.Status != model.StatusOutOfOffice {
-		c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
+		err = c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
+		if err != nil {
+			c.Err = err
+			return
+		}
 	}
```

Comment:
@hanzei - what are your thoughts on changing this to be more atomic?

```go
currentStatus, appErr := c.App.GetStatus(c.Params.UserId)
if appErr != nil {
  c.Err = appErr
  return
}
if currentStatus.Status == model.StatusOutOfOffice && status.Status != model.StatusOutOfOffice {
  appErr = c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
  if appErr != nil {
	c.Err = appErr
	return
  }
}
```

Reply:
Is it worth blocking the execution of the later code if `App.GetStatus` fails? I wonder what the side affects are if `App.GetStatus` continuously fails and the status doesn't get updated.

Reply:
If not, we should atleast log every error. Here, we are only executing the code if `appErr == nil` completely ignoring the fact that if `app.GetStatus` fails, there is no indication whatsoever as to why `DisableAutoResponder` was not executed.

Reply:
I'm fully supportive of logging the error :+1:

Reply:
And the error is logged!

```go
if err == nil && currentStatus.Status == model.StatusOutOfOffice && status.Status != model.StatusOutOfOffice {
		err = c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
		if err != nil {
			c.Err = err
			return
		}
        }
```

Do you guys @hanzei @agnivade mean something different?

Reply:
If `c.App.GetStatus` error, we don't log the error.

Reply:
@RS-labhub - Please go through the above discussion again. We want to log the error from `c.App.GetStatus(c.Params.UserId)`. The code you posted above doesn't do that.

Reply:
Got it!! Pushing the changes soon!

Reply:
You have pushed my original suggestion, which is not what we want. We do not want to return in case of error from `GetStatus`, but just log it.

If you have any confusion, feel free to reach out to me on community.mattermost.com at agniva.de.sarker.

Reply:
@agnivade sorry the confusion, could you check it now?